include/wlr/types/wlr_layer_shell_v1.h

```diff
@@ -71,15 +73,13 @@ struct wlr_layer_surface_v1 {
 	char *namespace;
 
 	bool added, configured, mapped;
-	uint32_t configure_serial;
-	uint32_t configure_next_serial;
 	struct wl_list configure_list;
 
-	struct wlr_layer_surface_v1_configure *acked_configure;
+	struct wlr_layer_surface_v1_state current, pending;
 
-	struct wlr_layer_surface_v1_state client_pending;
-	struct wlr_layer_surface_v1_state server_pending;
-	struct wlr_layer_surface_v1_state current;
+	// Properties to be sent to the client in the next configure event.
+	struct wlr_layer_surface_v1_configure scheduled;
```

Comment:
Hmm, so xdg-toplevel configuring is incremental in case it will be extended?
Maybe it will make sense to choose one way (immediate configuring / scheduled) and stick to that.

Reply:
Actually, I don't think we need to store the configure serial used either. Instead we should return it from `wlr_layer_surface_v1_configure()` in case the compositor wants to make use of it.

Reply:
> Hmm, so xdg-toplevel configuring is incremental in case it will be extended?
> Maybe it will make sense to choose one way (immediate configuring / scheduled) and stick to that

For xdg toplevel there is a lot more information included in a configure than just the width/height. This means that if a compositor calls `wlr_xdg_toplevel_set_fullscreen()` and then `wlr_xdg_toplevel_set_tiled()` before yielding back to the event loop we only want to send a single configure event not 2 separate ones.

With layer shell, the configure only carries the `width`/`height` which are both set using `wlr_layer_surface_v1_configure()` so we don't have this issue.

Reply:
An uber-function like `wlr_xdg_toplevel_configure(toplevel, width, height, resizing, activated, maximized, tiled)` could be created, but I guess it would be cumbersome to use.
I'm just not a huge fan of how the seemingly same configure mechanism is handled differently for different objects.

Reply:
I suppose there is an argument to be made for using a similar scheduling mechanism for layer shell as well:

If the compositor makes multiple `wlr_layer_surface_v1_configure()` calls with different width/height arguments before yielding back to the event loop we will currently sent multiple configure events instead of just the lastest one. I'm not sure if the slightly increased complexity of an `configure_idle` event source and callback is worth it or not to fix that, or if this is even a real issue in practice. 

@emersion thoughts?

Reply:
Added a commit which implements scheduling.